src/tqdm_publisher/_demos/_parallel_bars/_server.py

```diff
@@ -24,20 +24,24 @@
 # Each outer entry is a list of 'tasks' to perform on a particular worker
 # For demonstration purposes, each in the list of tasks is the length of time in seconds
 # that each iteration of the task takes to run and update the progress bar (emulated by sleeping)
-SECONDS_PER_TASK = 1
-NUMBER_OF_TASKS_PER_JOB = 6
+BASE_SECONDS_PER_TASK = 0.5  # The base time for each task; actual time increases proportional to the index of the task
+NUMBER_OF_TASKS_PER_JOB = 10
 TASK_TIMES: List[List[float]] = [
-    [SECONDS_PER_TASK * task_index] * task_index for task_index in range(1, NUMBER_OF_TASKS_PER_JOB + 1)
+    [BASE_SECONDS_PER_TASK * task_index] * NUMBER_OF_TASKS_PER_JOB
+    for task_index in range(1, NUMBER_OF_TASKS_PER_JOB + 1)
 ]
 
 WEBSOCKETS = {}
 
-## NOTE: TQDMProgressHandler cannot be called from a process...so we just use a queue directly
+## NOTE: TQDMProgressHandler cannot be called from a process...so we just use a global reference exposed to each subprocess
 progress_handler = TQDMProgressHandler()
 
 
-def forward_updates_over_sse(request_id, id, n, total, **kwargs):
-    progress_handler._announce(dict(request_id=request_id, id=id, format_dict=dict(n=n, total=total)))
+def forward_updates_over_server_side_events(request_id: str, progress_bar_id: str, n: int, total: int, **kwargs):
+    # TODO: shouldn't this line use `create_progress_subscriber`? Otherwise consider making `.accounce` non-private
```

Comment:
@garrettmflynn attn

Reply:
So `create_progress_subscriber` would require direct access to the same `progress_handler` across processes—which AFAIK we cannot do. Though we could swap to `create_progress_subscriber` in the Multiple Bars demo if we'd like to enforce serial message handling as that's all in one process (but multiple threads).

Providing `_announce` as a "private" property here is simply to indicate that the recommended usage is to use `create_progress_subscriber` (as noted in new docs) instead of direct use. Though the latter may still be useful in special cases.

I'm open to whatever you'd prefer.

Reply:
If the method can be that general, then I say just go ahead and make it public rather than private

Reply:
Updated